pkg/handler/project/project.go

```diff
@@ -132,3 +133,107 @@ func (h *handler) UpdateProjectStatus(c *gin.Context) {
 	// 3. return project data
 	c.JSON(http.StatusOK, view.CreateResponse[any](view.ToUpdateProjectStatusResponse(project), nil, nil, nil))
 }
+
+// UpdateProjectStatus godoc
+// @Summary Update status for project by id
+// @Description Update status for project by id
+// @Tags Project
+// @Accept json
+// @Produce json
+// @Param id path string true "Project ID"
+// @Param Body body CreateMemberInput true "Body"
+// @Param Authorization header string true "jwt token"
+// @Success 200 {object} view.CreateMemberDataResponse
+// @Failure 400 {object} view.ErrorResponse
+// @Failure 404 {object} view.ErrorResponse
+// @Failure 500 {object} view.ErrorResponse
+// @Router /projects/{id}/status [post]
+func (h *handler) AddMemberToProject(c *gin.Context) {
+	var body CreateMemberInput
+	if err := c.ShouldBind(&body); err != nil {
+		if err != nil {
+			c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, err, body))
+			return
+		}
+	}
+
+	projectID := c.Param("id")
+	if projectID == "" || !model.IsUUIDFromString(projectID) {
+		c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, errors.New("id is invalid"), nil))
+		return
+	}
+
+	// TODO: can we move this to middleware ?
+	l := h.logger.Fields(logger.Fields{
+		"handler": "project",
+		"method":  "AddMemberToProject",
+		"body":    body,
+	})
+
+	if err := body.Validate(); err != nil {
+		l.Error(err, "validate failed")
+		c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, err, body))
+		return
+	}
+
+	var seniorityID model.UUID
+	if body.SeniorityID != "" {
+		seniorityID = model.MustGetUUIDFromString(body.SeniorityID)
+	}
+	// create project slot
```

Comment:
I'm not sure but I think we should check project ID exists and return 404(not found) error before creating slot for it, because if we dont check, the error will be 500(Internal error)

Reply:
updated